pkg/segment/writer/segstore.go

```diff
@@ -938,18 +948,28 @@ func (segstore *SegStore) computeStarTree() {
 	}
 
 	if segstore.numBlocks != 0 {
-		cname, found, cardinality := segstore.isAnyAtreeColAboveCardLimit()
-		if found {
-			// todo when we implement dropping of columns from atree,
-			// drop the column here and remove the dropping of segtree
-			log.Errorf("computeStarTree: Release STB, found cname: %v with high card: %v, blockNum: %v",
-				cname, cardinality, segstore.numBlocks)
-
+		highCardCols, highCardVals := segstore.getColsAboveCardLimit()
+		activeCols := segstore.stbHolder.stbPtr.numOfActiveCols()
+		if int(activeCols)-len(highCardCols) <= 0 {
+			log.Warnf("computeStarTree: Dropping SegTree All remaining cols %v found with high card: %v, blockNum: %v",
+				highCardCols, highCardVals, segstore.numBlocks)
 			segstore.stbHolder.stbPtr.DropSegTree(segstore.stbDictEncWorkBuf)
 			segstore.stbHolder.ReleaseSTB()
 			segstore.stbHolder = nil
 			return
 		}
+		for idx, cname := range highCardCols {
+			log.Warnf("computeStarTree: Dropping column, found cname: %v with high card: %v, blockNum: %v",
+				cname, highCardVals[idx], segstore.numBlocks)
```

Comment:
nit: for log messages, it would be good to be a little more clear
```suggestion
			log.Warnf("computeStarTree: Dropping column, found cname: %v with high cardinality: %v, blockNum: %v",
				cname, highCardVals[idx], segstore.numBlocks)
```

Same for lines 954-955